spec/plurimath/mathml/parse_spec.rb

```diff
@@ -0,0 +1,166 @@
+        <math xmlns='http://www.w3.org/1998/Math/MathML'>
+          <mstyle displaystyle='true'>
+            <munderover>
+              <mo>&#x2211;</mo>
+              <mrow>
+                <mo>(</mo>
+                <mo>&#x220f;</mo>
+                <mo>)</mo>
+              </mrow>
+              <mo>&#x22c1;</mo>
+            </munderover>
+          </mstyle>
+        </math>
+      MATHML
+    }
+    it "returns formula of sum and prod" do
+      expected_value = {
+                        :open=>"math",
+                        :attributes=>[{:name=>"xmlns", :value=>"http://www.w3.org/1998/Math/MathML"}],
+                        :iteration=>
+                        {:tag=>
+                          {:open=>"mstyle",
+                           :attributes=>[{:name=>"displaystyle", :value=>"true"}],
+                           :iteration=>
+                            {:tag=>
+                              {:open=>"munderover",
+                               :attributes=>[],
+                               :iteration=>
+                                {:tag=>{:open=>"mo", :attributes=>[], :iteration=>{:symbol=>"&#x2211;"}, :close=>"mo"},
+                                 :sequence=>
+                                  {:tag=>
+                                    {:open=>"mrow",
+                                     :attributes=>[],
+                                     :iteration=>
+                                      {:tag=>{:open=>"mo", :attributes=>[], :iteration=>{:symbol=>"("}, :close=>"mo"},
+                                       :sequence=>
+                                        {:tag=>{:open=>"mo", :attributes=>[], :iteration=>{:symbol=>"&#x220f;"}, :close=>"mo"},
+                                         :sequence=>{:tag=>{:open=>"mo", :attributes=>[], :iteration=>{:symbol=>")"}, :close=>"mo"}}},
+                                       :iteration=>""},
+                                     :close=>"mrow"},
+                                   :sequence=>{:tag=>{:open=>"mo", :attributes=>[], :iteration=>{:symbol=>"&#x22c1;"}, :close=>"mo"}}},
+                                 :iteration=>""},
+                               :close=>"munderover"},
+                             :iteration=>""},
+                           :close=>"mstyle"},
+                         :iteration=>""},
+                        :close=>"math"
+                      }
+      expect(formula).to eq(expected_value)
+    end
+  end
+
+  context "contains mathml string of sum formula" do
+    let(:exp) {
+      <<~MATHML
+        <math xmlns='http://www.w3.org/1998/Math/MathML'>
+          <mstyle displaystyle='true'>
+            <mrow>
+              <munderover>
+                <mo>&#x2211;</mo>
+                <mi>x</mi>
+                <mi>s</mi>
+              </munderover>
+            </mrow>
+          </mstyle>
+        </math>
+      MATHML
+    }
+    it "returns formula of sum and prod" do
+      expected_value = {
```

Comment:
Is it possible to make this comparison more simple? For example, if the structure changes in the future, these compares will be very tough to update. You should be testing for particular outcomes, not data equivalence of the structures.

Reply:
But we do do that in a future PR, too.